crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/class_info.rs

```diff
@@ -275,6 +279,163 @@ mod get_utility_info_tests {
     }
 }
 
+// variants
+// -------
+
+/// The result of matching a variant against a target.
+#[derive(Debug, Eq, PartialEq)]
+enum VariantMatch {
+    /// The variant matches an exact target.
+    Exact,
+    /// The variant matches a partial target.
+    Partial,
+    /// The variant does not match the target.
+    None,
+}
+
+impl VariantMatch {
+    /// Checks if a variant matches a target, and returns the result.
+    fn from(target: &str, variant_text: &str) -> VariantMatch {
+        // If the target matched exactly the variant text.
+        if target == variant_text {
+            return VariantMatch::Exact;
+        };
+
+        // if it has a custom value thus it starts with the variant text and it's followed by "-["
+        if variant_text.starts_with(format!("{}-[", target).as_str()) {
+            return VariantMatch::Exact;
+        }
+
+        // Check if the variant starts with the (partial) target.
+        if variant_text.starts_with(target) && variant_text != target {
+            return VariantMatch::Partial;
+        }
+        // If all of the above checks fail, there is no match.
+        VariantMatch::None
+    }
+}
+
+#[cfg(test)]
+mod variant_match_tests {
+    use crate::lint::nursery::use_sorted_classes::class_info::VariantMatch;
+
+    #[test]
+    fn test_exact_match() {
+        assert_eq!(VariantMatch::from("hover", "hover"), VariantMatch::Exact);
+        assert_eq!(VariantMatch::from("focus", "focus"), VariantMatch::Exact);
+        assert_eq!(
+            VariantMatch::from("group", "group-[.is-published]"),
+            VariantMatch::Exact
+        );
+        assert_eq!(
+            VariantMatch::from("has", "has-[:checked]"),
+            VariantMatch::Exact
+        );
+        assert_eq!(
+            VariantMatch::from("group-has", "group-has-[.custom-class]"),
+            VariantMatch::Exact
+        );
+        assert_eq!(
+            VariantMatch::from("group-aria-disabled", "group-aria-disabled"),
+            VariantMatch::Exact
+        );
+    }
+
+    #[test]
+    fn test_partial_match() {
+        assert_eq!(
+            VariantMatch::from("group", "group-has-[.custom-class]"),
+            VariantMatch::Partial
+        );
+        assert_eq!(
+            VariantMatch::from("peer", "peer-has-[:checked]"),
+            VariantMatch::Partial
+        );
+    }
+}
+
+fn find_variant_info(config_variants: VariantsConfig, variant_text: &str) -> Option<usize> {
```

Comment:
nit: since this function returns a number, maybe we could give `_info` a better name. In other words, what's "info"? What does the number we return represent?